frontend/app/src/components/Sidebar/styled-components.ts

```diff
@@ -21,6 +21,17 @@ import {
   hasLightBackgroundColor,
 } from "@streamlit/lib/src/theme/utils"
 import { StyledMaterialIcon } from "@streamlit/lib/src/components/shared/Icon/Material/styled-components"
+
+// Check for custom text color & handle colors in Sidebar/SidebarNav accordingly
+const hasCustomTextColor = (theme: any): boolean => {
```

Comment:
One other idea to consider is whether it should handle the ternary

like what if the function signature would be:

```js
const conditionalColorByCustomText = (theme, customColor, fallbackColor)
```

That would further reduce the ternaries.

Not blocking, but something to consider if it cleans it up further.